src/test/java/org/springframework/data/redis/connection/lettuce/LettuceConnectionFactoryTests.java

```diff
@@ -250,4 +254,28 @@ public void testConnectWithPassword() {
 		conn.bLPop(1, "key".getBytes());
 		conn.close();
 	}
+
+	/**
+	 * @see DATAREDIS-431
+	 */
+	@Test
+	public void dbIndexShouldBePropagatedCorrectly() {
+
+		LettuceConnectionFactory factory = new LettuceConnectionFactory();
+		factory.setDatabase(2);
+		factory.afterPropertiesSet();
```

Comment:
The `LettuceConnectionFactory` stays open which will not `shutdown()` the `RedisClient` -> Threads will stay alive and eat up your resources. 

lettuce 3.4 will introduce reusable thread pools, that might come hand for tests and the connection factory.